component/registry.go

```diff
@@ -31,8 +31,10 @@ var (
 
 // ModuleController is a mechanism responsible for allowing components to create other components via modules.
 type ModuleController interface {
-	// NewModule creates a new, un-started Module.
-	NewModule(id string, export ExportFunc) Module
+	// NewModule creates a new, un-started Module. The id must be unique to the ModuleController.
+	// ID is appended to the global id of the calling component, and can be left blank.
+	// Though any id used needs to be unique for that component.
```

Comment:
WDYT about this re-write? I tried to write in in a way so the uniqueness of id only needs to be mentioned once:

```suggestion
	// NewModule creates a new, un-started Module with a given ID. Multiple calls to 
	// NewModule must provide unique values for id. The empty string is a valid unique 
	// value for id. 
```